projects/@crucible-common/src/lib/comn-auth/services/comn-auth.service.ts

```diff
@@ -91,6 +119,59 @@ export class ComnAuthService {
   private onTokenLoaded(user) {
     this.user = user;
     this.store.update({ user });
-    const userGuid = user.profile.sub;
+    // set inactivity and token expiration monitoring
+    this.setExpirationTimer();
+    this.startInactivityMonitor();
+  }
+
+  startInactivityMonitor(): void {
+    // No inactivityTimeSeconds means inactivity monitor is disabled
+    if (this.inactivityTimeSeconds) {
+      this.ngZone.runOutsideAngular(() => {
+        this.activityObserveable$ = this.mergedEventObservable$
+        .pipe(
+          switchMap(ev => interval(1000).pipe(take(this.inactivityTimeSeconds))),
+          skipWhile((x) => {
+            this.timeLapsedSinceInactivity = x;
+            return x != this.inactivityTimeSeconds - 1
+          })
+        );
+        this.subscribeObservable();
+        this.kickstartObservable$.next(true);
+      });
+    }
   }
+
+  subscribeObservable() {
+    this.inactivitySubscription = this.activityObserveable$.subscribe((x) => {
+      // check for a configured inactivity redirect url
+      if (this.settingsService.settings.inactivityRedirectUrl)
+      {
+        // goto the configured url
+        this.inactivitySubscription.unsubscribe();
+        document.location.href = this.settingsService.settings.inactivityRedirectUrl;
+      }
+      else
+      {
+        // goto the signout redirect page
+        this.inactivitySubscription.unsubscribe();
+        this.logout();
+      }
+    });
+  }
+
+  setExpirationTimer() {
```

Comment:
Can we make this configurable through a setting as well? I'm not sure if it will cause any issues in environments with shorter access token expirations and/or other identity providers

Reply:
Great idea!  Will do!  Thanks!  It will be disabled by default.